modules/java/src/main/java/org/syslog_ng/LogDestination.java

```diff
@@ -56,8 +56,8 @@ public int getSeqNum() {
 
 	private native int getSeqNum(long ptr);
 
-	protected void onMessageQueueEmpty() {
-		return;
+	protected boolean flush() {
```

Comment:
I understand that we want to change the interface here, and without deprecating the old one. (I think we could even get away with it.)
But if we accept this change, does this change enough ?
Are we going to feel that the java interface is good enough for at least mid term ?
I think we should hurt the user once, and not continuously.